lib/hash.c

```diff
@@ -2340,19 +2353,28 @@ grn_hash_add(grn_ctx *ctx, grn_hash *hash, const void *key,
     const uint32_t step = grn_hash_calculate_step(hash_value);
     grn_id id, *index, *garbage_index = NULL;
     grn_hash_entry *entry;
+    grn_rc rc;
 
     /* lock */
     if ((*hash->n_entries + *hash->n_garbages) * 2 > *hash->max_offset) {
       if (*hash->max_offset > (1 << 29)) {
         ERR(GRN_TOO_LARGE_OFFSET, "hash table size limit");
         return GRN_ID_NIL;
       }
-      grn_hash_reset(ctx, hash, 0);
+      rc = grn_hash_reset(ctx, hash, 0);
+      if (rc != GRN_SUCCESS) {
+        ERR(rc, "[hash][add] failed to reset hash");
+      }
     }
 
     for (i = hash_value; ; i += step) {
       index = grn_hash_idx_at(ctx, hash, i);
       if (!index) {
+        char name[GRN_TABLE_MAX_KEY_SIZE];
+        int name_size;
+        name_size = grn_hash_name(ctx, hash, name, GRN_TABLE_MAX_KEY_SIZE);
+        ERR(GRN_UNKNOWN_ERROR, "[hash][add] failed to detect index of <%.*s> at(%d)",
```

Comment:
Could you collect all values at the last instead of embedding values in message? It's easy to debug.
`failed to detect index: <%.*s>: <%u>`